### PR TITLE
fix(388): replace zerolog reference with slog

### PR DESCRIPTION
**Pull request by lsytj0413**
Fixes: https://github.com/streamnative/oxia/issues/388

### CODE REVIEW

File: common/client_pool.go [hunk 1]
```diff
@@ -65,10 +64,11 @@ func (cp *clientPool) Close() error {
...
 		err := cnx.(*grpc.ClientConn).Close()
 		if err != nil {
-			cp.log.Warn().
-				Str("server_address", target).
-				Err(err).
-				Msg("Failed to close GRPC connection")
+			cp.log.Warn(
+				"Failed to close GRPC connection",
+				slog.String("server_address", target),
+				slog.Any("Error", err),
```
Comment (merlimat) [inline common/client_pool.go:70]: nit: should we consider standardizing on `error` as the key for all error messages?

**Review by merlimat — APPROVED** [review]
Thanks for the great refactoring!

File: common/client_pool.go (hunk 1 above)
Comment (lsytj0413) [reply common/client_pool.go:70]: yes，i name the key of error as `Error`，and slog didn't provide an convention for it.

Comment (merlimat) [reply common/client_pool.go:70]: I was meaning: would it be better to use lower cap `error` to be more consistent to the other log keys?

Comment (lsytj0413) [reply common/client_pool.go:70]: sorry, i misunderstand it. I will fix it at next time.

**Pull request merged**